napytau/gui/components/graph.py

```diff
@@ -28,57 +40,171 @@ def update_plot(self) -> None:
         Updates the graph with the latest tau value and appearance mode.
         """
         self.graph_frame = self.plot(
-            self.parent.tau.get(), customtkinter.get_appearance_mode()
+             customtkinter.get_appearance_mode()
         )
         self.graph_frame.grid(
             row=0, column=0, rowspan=2, padx=10, pady=10, sticky="nsew"
         )
         self.graph_frame.grid_propagate(False)
 
-    def plot(self, value: int, appearance: str) -> Canvas:
+    def plot(self, appearance: str) -> Canvas:
         """
         Plot the graph.
         :param appearance: The appearance mode.
-        :param value: The value.
         :return: The canvas.
         """
-        # the figure that will contain the plot
-        fig = Figure(figsize=(3, 2), dpi=100, facecolor="white", edgecolor="black")
-
-        # setting color in dependence of appearance mode
-        if appearance == "Light":
-            main_color = "white"
-            secondary_color = "#000000"
-        else:
-            main_color = "#151515"
-            secondary_color = "#ffffff"
 
-        fig.patch.set_facecolor(main_color)
+        # Placeholder datapoints:
+        y_data = [2, 4, 5, 7, 8, 9, 11, 15]
+        distances = [1, 2, 3, 4, 5, 6, 7, 8]
+        x_data = [0, 1, 2, 3, 4, 5, 6, 7]
 
```

Comment:
I think the placeholder values should be given to `def plot(self, appearance: str) -> Canvas:` as parameters and not be hard-coded in the plot function itself.
Additionaly there are already placeholder values saved in `class App(customtkinter.CTk):`
```
# Datapoints
self.datapoints: List[Tuple[float, float]] = []
self.datapoints_for_fitting: List[CheckboxDataPoint] = []
self.datapoints_for_calculation: List[CheckboxDataPoint] = []
```
The `data_points_for_fitting` list stores the data points that the user selects (via the checkboxes) which should be used for fitting. So the graph should use the points from this list.
The class `CheckboxDataPoint` would probably need to be expanded in terms of the error value of each datapoint.